src/psyclone/psyir/nodes/node.py

```diff
@@ -1149,6 +1149,40 @@ def scope(self):
             "Unable to find the scope of node '{0}' as none of its ancestors "
             "are Container or Schedule nodes.".format(self))
 
+    def replace_with(self, node):
+        '''Replaces self with the supplied node in the PSyIR tree.
```

Comment:
This sentence seems ambiguous. Maybe: "Replaces self, and all its children, from the PSyIR tree where it belongs with with supplied node."

Reply:
Updaed.